torch/lib/c10d/TCPStore.cpp

```diff
@@ -522,6 +726,30 @@ bool TCPStore::deleteKey(const std::string& key) {
   return (numDeleted == 1);
 }
 
+void TCPStore::watchKey(
+    const std::string& key,
+    StoreCallbackFunction callback) {
+  // Only allow one thread to perform watchKey() at a time
+  const std::lock_guard<std::mutex> watchKeyLock(watchKeyMutex);
```

Comment:
curious, what is the motivation of having two mutex/locks? (Another one is `keyToCallbacksMutex`)

Reply:
keyToCallbacksMutex is for the map on TCPStoreWorkerDaemon and watchKeyMutex is for the operation of registering on masterDaemon and workerDaemon

Reply:
Trying to understand the benefit of having two locks instead of using one for both cases. Is the main motivation to allow `TCPStoreWorkerDaemon::callbackHandler` to run concurrently with this function as much as possible?

Reply:
addressed offline.